Develop

simarbir opened the pull request:
Initial commit for Cisco Meraki vMX and CloudWAN integration.

simarbir committed:
Intial commit

simarbir committed:
Start using vmx-tgw qs submodule

simarbir committed:
Fix up

simarbir committed:
Remove .DS_Store

simarbir committed:
Clean up source directory. Remove dependencies.

simarbir committed:
clean up

simarbir committed:
Add additional region statemachine

mufaddalq committed:
inital commit for additional region support

simarbir committed:
Update lambdas to use eventbus

simarbir committed:
Merge pull request #1 from mufaddalq/feature/additional-region

inital commit for additional region support

simarbir committed:
Fix up

simarbir committed:
Additional region integration updates

simarbir committed:
Move create_step_function lambda to inline

jleatham committed:
added inline lambdas and step function

simarbir committed:
Merge pull request #2 from jleatham/develop

added inline lambdas and step function

mufaddalq committed:
plumbing end-to-end delete

mufaddalq committed:
create network plumbing

mufaddalq committed:
deleting all functions that are now inline

mufaddalq committed:
fixed typo

simarbir committed:
Merge pull request #3 from mufaddalq/temp2

Temp2

mufaddalq committed:
expressing stepfunction invocation event as cloudformation and setting it as a dependency on the custom resource

jleatham committed:
added delete stack and moved resources

jleatham committed:
fixed some EventBusName references

mufaddalq committed:
more modificaiton to get multi-region work

simarbir committed:
Merge branch 'develop' into temp2

simarbir committed:
Merge pull request #4 from mufaddalq/temp2

Temp2

simarbir committed:
Updated polling lambda to send VpcAttachmentId and CoreNetworkId

simarbir committed:
update create statemachine

jleatham committed:
fixed another EventBusName references

jleatham committed:
fixed linting errors

jleatham committed:
fixed linting errors

mufaddalq committed:
updating logic for create_network_policy function based on latest polling lanbda changes

simarbir committed:
Merge pull request #5 from mufaddalq/temp2

updating logic for create_network_policy function based on latest pol…

jleatham committed:
callback function permissions

jleatham committed:
Merge branch 'develop' of https://github.com/simarbir/quickstart-cisco-meraki-vmx-cloudwan into develop

jleatham committed:
added back delete event

jleatham committed:
added tag to global network for delete

jleatham committed:
syntax bugs and custom resource changes

simarbir committed:
Fix ups

jleatham committed:
fixed VPCAttachment python error + small additions elsewhere

jleatham committed:
vendored requests put doesn't work in various regions

jleatham committed:
subnetARNs were double listed

jleatham committed:
fixed VPCAttachment issues - need to find core network based upon global network.  Can't be passed in?

jleatham committed:
copylambdas pointing to wrong file - caused api deny error

jleatham committed:
fixed  output name error

jleatham committed:
custom resource typo fix

jleatham committed:
Duplicate perameter removed

jleatham committed:
commented out non existant outputs

simarbir committed:
Polling lambda fixes and start matching on network_name tag for global networks

simarbir committed:
Polling lambda fixes and updates to wait times for delete state machine

jleatham committed:
yet another output error fixed

jleatham committed:
commented out some unused params

jleatham committed:
added ASN param back in as a dummy value is used

jleatham committed:
fixed globalnetwork param not being passed

jleatham committed:
simple fix to move return inside try statement

mufaddalq committed:
fixing ASNRange, BaseRegionName, GlobalNetworkName and MerakiEventBusName Params for additional region

jleatham committed:
improved state machines to cover race conditions and add edge locations for new region before attachment creation

jleatham committed:
added '' to 'network_name'

jleatham committed:
fixed the attachment logic for delete

jleatham committed:
CreateNewRegion State Machine had hardcoded ARNS

jleatham committed:
fixed typo

jleatham committed:
fixed typo

jleatham committed:
base region was being added twice to policy

jleatham committed:
small change

jleatham committed:
fixing logic for both base region and additional regions - edge locations

jleatham committed:
Continued fixing logic for VpcCreation StateMachine

jleatham committed:
re-added params removed on accident

jleatham committed:
name typo fixed...again

jleatham committed:
fixed update policy logic to check if policy changed

jleatham committed:
removed another output param

jleatham committed:
started update function tshoot - not finished

simarbir committed:
Fix polling lambda to send strings for routes

simarbir committed:
Update polling lambda to not call the update state machine
when there is no route change.

simarbir committed:
Remove asn-range ref from lambda template and polling lambda

jleatham committed:
removed useless output

jleatham committed:
added default for ASN until we remove it altogether

jleatham committed:
added success exit for update state machine

simarbir committed:
Merge branch 'develop' into josh_dev

simarbir committed:
Merge pull request #6 from simarbir/josh_dev

Josh dev

jleatham committed:
fixed issue where region was added to policy multiple times

simarbir committed:
Fix ups for update network policy function and polling lambda

sshvans reviewed (commented):
There seems to be some open questions from previous comments, which needs to be addressed.
Can we also remove commented code. As it adds no value, other than where there are useful comments.

simarbir committed:
Update polling lambda to handle route updates

jleatham committed:
added the ability to remove segment-actions if no cidrs are found

simarbir committed:
Fix up for polling lambda

simarbir committed:
Fix up

jleatham committed:
added to the UpdateStateMachine logic

jleatham committed:
cleaned up the comments/descriptions

jleatham committed:
Fixed name tag to Name for CW attachment policy

jleatham committed:
polling lambda- fixed for loop errors

jleatham committed:
remove network logic bug - ran during createNewNetwork update and throws error

sshvans committed:
Removing auth configuration

Removing auth configuration from taskcat file to enable ci to use default auth. If you need explicit auth profile, add it to your override file.

simarbir committed:
Remove template.yaml as it is no longer used

simarbir committed:
Fixes for bandit checks: B105 and B110

sshvans committed:
Create .metadata

Needed by QS CI
